ucb-rit/coldfront
# Update API endpoints for cluster access request processing; add UserProfile history

Refs #382

**Changes**
- Included `user_id` in `AllocationUserSerializer` to allow further lookups of user data.
- Added `UserProfileSerializer` and included it as `profile` in `UserSerializer` to include `UserProfile` information in the API.
    - Only included the following fields for LRC: `billing_activity`, `host_user`, `host_user_lbl_email`.
- Added a `history` field to the `UserProfile` method to track changes to `billing_activity` and `host_user`.
- Added and updated API endpoint tests.
- Moved utility methods relating to host users from `user.utils` to `user.utils_.host_user_utils`.
- Moved statements out of the transaction in `ClusterAccessRequestViewSet`.
- Updated fields displayed in the list view for `ClusterAccessRequests`.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes.
- Manual Testing
    - On an LRC VM:
        - Create a `ClusterAccessRequest`.
        - Manually set its `billing_activity` and/or `host_user` as needed.
        - Ensure that the correct values are included in the list view and in the API.
        - Ensure that the API includes a `user_id`.
        - Navigate to the corresponding `/api/users/{id}/` endpoint, and ensure that it includes a `profile` field with the correct values.
